### ISSUE TITLE
[PHP 7.1] 4/3/19 Update causing php error from Line 136 in api.php

### ISSUE THREAD

**Issue by adrianwaler** [mgibbs189/custom-field-suite#455]
```PHP Fatal error: Uncaught Error: [] operator not supported for strings in /wp-content/plugins/custom-field-suite/includes/api.php:136```

**Comment by mgibbs189** [comment]
@adrianwaler add the following directly above that line:

```php
echo '<pre>';
var_dump( $field_data );
var_dump( $hierarchy );
echo '</pre>';
```

Can you show me what that outputs?

**Comment by adrianwaler** [comment]
Sadly I'm not able to setup this project locally that fast. As I remember from debugging issue some time ago `$field_data[ $hierarchy ]` was empty string.  Because of that
```
if ( ! isset( $field_data[ $hierarchy ] ) ) {
    $field_data[ $hierarchy ] = array();
}
```
is not executed. If it's needed I can try to setup this project but I will be back with  result in next week. Error from issue description is from server logs.

**Comment by adrianwaler** [comment]
@mgibbs189 Last output before error:
```
array(7) {
  ["post_image_big"]=>
  string(0) ""
  ["post_featured"]=>
  int(0)
  ["post_layout"]=>
  array(1) {
    ["typical"]=>
    string(7) "typical"
  }
  ["data_layer_route"]=>
  string(15) "media"
  ["content_grouping_enabled"]=>
  int(0)
  ["content_grouping"]=>
  array(1) {
    [1]=>
    string(9) "Awareness"
  }
  ["content_group"]=>
  array(1) {
    [0]=>
    string(0) ""
  }
}
string(18) "post_image_big"
```

**Comment by adrianwaler** [comment]
@mgibbs189 Fix proposed by @thenomadicmann in works well:
```
                        if ( isset( $field_data[ $hierarchy ] ) && is_array( $field_data[ $hierarchy ] ) ) {
                            $field_data[ $hierarchy ][] = $result->meta_value;
                        } else {
                            $field_data[ $hierarchy ] = [$result->meta_value];
                        }
```

Please, could you fix that? Hosting forced me to use php 7.1 so I have to maintain this fix on my own, which is not best approach.
